StencilFrontend.md

```diff
@@ -0,0 +1,353 @@
+
+{% for meal in meals %}
+    - {{ meal.name }} with rating {{ meal.rating }}. <br />
+    <br />
+{% endfor %}
+</html>
+```
+5. Save the file and exit your text editor. For `cat` use the shortcut:  `CTRL+D`
+
+When the above code is rendered, it will display the total number of meals in your meal store, it will then loop through the meal store displaying each meal name and rating. For more details about Stencil templates see the [Stencil User Guide](http://stencil.fuller.li/en/latest/).
+
+
+### Add the Kitura-StencilTemplateEngine dependency
+[Kitura-TemplateEngine](https://github.com/IBM-Swift/Kitura-TemplateEngine) is a template engine abstraction layer which works alongside a specific template engine, such as [Kitura-StencilTemplateEngine](https://github.com/IBM-Swift/Kitura-StencilTemplateEngine), to allow a user to easily use a template engine in Swift. We need to add the `Kitura-StencilTemplateEngine` dependency to our `Package.swift` file.
+
+1. In the terminal, go to your server's `Package.swift` file.
+```
+cd ~/FoodTrackerBackend/FoodServer
+open Package.swift
+```
+2. Add the `Kitura-StencilTemplateEngine` package:
+```swift
+.package(url: "https://github.com/IBM-Swift/Kitura-StencilTemplateEngine.git", from: "1.8.0")
+```
+3. Change the target for Application to include "KituraStencil":
+```swift
+.target(name: "Application", dependencies: [ "Kitura", "Configuration", "CloudEnvironment", "SwiftMetrics", "Health", "KituraStencil"]),
+```
+
+### Generate your FoodServer Xcode project
+
+Now we have added the dependencies to our `Package.swift` file we can generate our FoodServer Xcode project to make editing the code easier. The FoodServer is a pure Swift project so the following steps could also be achieved by editing the `.swift` files.
+
+1. Generate and open the server Xcode project:
+```
+swift package generate-xcodeproj
+open FoodServer.xcodeproj/
+```
+2. Click on the "FoodServer-Package" text on the top-left of the toolbar and select "Edit scheme" from the dropdown menu.
+3. In "Run" click on the "Executable" dropdown, select FoodServer and click Close.
+
+Now, when you press play, Xcode will start your FoodTracker server listening on port 8080. You can see this by going to [http://localhost:8080/](http://localhost:8080/ ) which will show the default Kitura landing page.
+
+### Add the Stencil template engine
+1. Open your `Sources > Application > Application.swift` file
+2. Add KituraStencil to the import statements:
+```swift
+import KituraStencil
+```
+3. Add the template engine to the router.
+
+Inside the `postInit()` function, insert the following line, below `router.get("/meals", handler: loadHandler)`:
+```swift
+router.add(templateEngine: StencilTemplateEngine())
+```
+
+The router will now use a `StencilTemplateEngine()` to render templates. You can use multiple template engines in one file and `router.setDefault(templateEngine: StencilTemplateEngine())` can be used to set the default.
```

Comment:
"You can use multiple template engines in one file and `router.setDefault(templateEngine: StencilTemplateEngine())` can be used to set the default."
Although this is true is it useful to the user to know that here, as we're not using any of that information directly?